components/apimgt/org.wso2.carbon.apimgt.rest.api.admin/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/impl/TenantInfoApiServiceImpl.java

```diff
@@ -16,31 +16,44 @@
  */
 package org.wso2.carbon.apimgt.rest.api.admin.impl;
 
+import org.apache.commons.lang3.StringUtils;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.wso2.carbon.apimgt.api.APIManagementException;
 import org.wso2.carbon.apimgt.impl.utils.APIUtil;
 import org.wso2.carbon.apimgt.rest.api.admin.TenantInfoApiService;
 import org.wso2.carbon.apimgt.rest.api.admin.dto.TenantInfoDTO;
 import org.wso2.carbon.apimgt.rest.api.util.utils.RestApiUtil;
+import org.wso2.carbon.utils.multitenancy.MultitenantUtils;
 
 import javax.ws.rs.core.Response;
+import java.util.Base64;
 
 public class TenantInfoApiServiceImpl extends TenantInfoApiService {
     Log log = LogFactory.getLog(TenantInfoApiServiceImpl.class);
 
     @Override
     public Response getTenantInfoByUsername(String username){
         TenantInfoDTO tenantInfoDTO = new TenantInfoDTO();
+        String decodedUserName;
         try {
-            if (!APIUtil.isUserExist(username)) {
-                RestApiUtil.handleBadRequest("Requested User does not exist", log);
+            if (StringUtils.isEmpty(username)) {
+                RestApiUtil.handleBadRequest("User Name should not be empty", log);
             }
-            int tenantId = APIUtil.getTenantId(username);
-            String tenantDomain = APIUtil.getTenantDomainFromTenantId(tenantId);
+            try {
+                decodedUserName = new String(Base64.getDecoder().decode(username));
+            } catch (IllegalArgumentException e) {
+                log.warn("Could not decode the username. Using original username");
```

Comment:
better to log the error logs

Reply:
Fixed in 59acdeb002d24819ab5972776a7fe7e60251c59c

Reply:
Fixed wth 59acdeb002d24819ab5972776a7fe7e60251c59c